File: Android_Study/Study/MyCallIntent/app/src/main/java/com/satelite54/callintent/MainActivity.java
package com.satelite54.callintent;

import androidx.appcompat.app.AppCompatActivity;

import android.content.ComponentName;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button button = findViewById(R.id.button);

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse("tel:010-3484-5022"));
                startActivity(intent);
                //Intent는 엑티비티가 엑티비티 스택에 쌓일 때 옵션을 지정해줄 수 있다. flag를 사용해서
/*                Intent intent = new Intent();
                ComponentName name = new ComponentName("com.satelite54.callintent", "com.satelite54.callintent.MenuActivity"); //문자열을 이용해서 엑티비티를 호출할 수 있다.
                intent.setComponent(name);
                startActivityForResult(intent, 101);*/
            }
        });
    }
}

File: Android_Busan_it/MyLifecycle_Intent/app/src/main/java/com/example/mylifecycle_intent/MainActivity.kt
package com.example.mylifecycle_intent

import android.content.Intent
import android.net.Uri
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.PersistableBundle
import android.provider.ContactsContract.Intents.Insert.ACTION
import android.widget.Button

class MainActivity : AppCompatActivity() {
    lateinit var Call : Button
    lateinit var End : Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        android.util.Log.i("액티비티 테스트","onCreate()")
        Call = findViewById<Button>(R.id.Call)
        End = findViewById<Button>(R.id.End)

        Call.setOnClickListener {
            var intent = Intent(Intent.ACTION_DIAL, Uri.parse("tel:119"))
            startActivity(intent)
        }

        End.setOnClickListener {
            finish()
        }
    }

    override fun onStart() {
        super.onStart()
        android.util.Log.i("액티비티 테스트","onStart()")
    }

    override fun onCreate(savedInstanceState: Bundle?, persistentState: PersistableBundle?) {
        super.onCreate(savedInstanceState, persistentState)
        android.util.Log.i("액티비티 테스트", "onCreate()")
    }

    override fun onResume() {
        super.onResume()
        android.util.Log.i("액티비티 테스트", "onResume()")
    }

    override fun onPause() {
        super.onPause()
        android.util.Log.i("액티비티 테스트", "onPause()")
    }

    override fun onStop() {
        super.onStop()
        android.util.Log.i("액티비티 테스트", "onStop()")
    }

    override fun onDestroy() {
        super.onDestroy()
        android.util.Log.i("액티비티 테스트" ,"onDestroy()")
    }

    override fun onRestart() {
        super.onRestart()
        android.util.Log.i("액티비티 테스트", "onRestart()")
    }

    var dw = funInterface {
        System.out.println(121213213)
    }
}